### PR TITLE
Adds documentation for overriding default serializer based on conditions

### PR DESCRIPTION
**Pull request by cgmckeever**
#### Purpose

Conversation on 'dynamic serializers' spawned an interesting solution via @groyoh 
Just documenting it here as its a 💯 #protip  

#### Changes

Updated serializer docs

#### Related GitHub issues

https://github.com/rails-api/active_model_serializers/issues/1719

### CODE REVIEW

File: docs/general/serializers.md [hunk 1]
```diff
@@ -369,3 +369,19 @@ class PostSerializer < ActiveModel::Serializer
   end
 end
 ```
+
+## Overriding default serializer
+
+You can use the `serializer_for` method to return a different serializer based on
+use/scope specific conditions.
```
Comment (cgmckeever) [inline docs/general/serializers.md]: wording welcome!

File: docs/general/serializers.md [hunk 2]
```diff
@@ -369,3 +369,19 @@ class PostSerializer < ActiveModel::Serializer
   end
 end
 ```
+
+## Overriding default serializer
```
Comment (groyoh) [inline docs/general/serializers.md]: I'd rather change it to `## Overriding association serializer lookup` as this should only works for the relationships of this specific serializer.

File: docs/general/serializers.md [hunk 3]
```diff
@@ -369,3 +369,19 @@ class PostSerializer < ActiveModel::Serializer
...
+
+You can use the `serializer_for` method to return a different serializer based on
```
Comment (groyoh) [inline docs/general/serializers.md]: Can you specify that the `serializer_for` is a class method? What do you think of something like:

```diff
+If you want to define a specific serializer lookup for your associations, you can override the `ActiveModel::Serializer.serializer_for` method to return a serializer class based on the model .
```

?

**Comment by groyoh** [comment]
@cgmckeever thanks for this PR! :100:

File: docs/general/serializers.md [hunk 4]
```diff
@@ -369,3 +369,19 @@ class PostSerializer < ActiveModel::Serializer
...
+
+```ruby
+class MySerializer < ActiveModel::Serializer
+  def self.serializer_for(model, options)
+    return SparseAdminSerializer if model.class == 'Admin'
+    super
+  end
+
+  # the rest of the serializer
+end
```
Comment (cgmckeever) [inline docs/general/serializers.md]: @groyoh much better suggestion - updated

**Comment by NullVoxPopuli** [comment]
This is super handy to document!

**Comment by NullVoxPopuli** [comment]
LGTM

File: CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,7 @@ Fixes:
 
 Misc:
 - [#1673](https://github.com/rails-api/active_model_serializers/pull/1673) Adds "How to" guide on using AMS with POROs (@DrSayre)
+- [#1730](https://github.com/rails-api/active_model_serializers/pull/1730) Adds documentation for overriding default serializer based on conditions (@groyoh/@cgmckeever)
```
Comment (cgmckeever) [inline CHANGELOG.md]: @NullVoxPopuli added

Comment (NullVoxPopuli) [inline CHANGELOG.md]: wooooo

**Comment by KevinColemanInc** [comment]
@groyoh hey! so I don't think this documentation is correct.  I tried implementing this and it doesn't seem to have the desired effect.

    class MySerializer < ActiveModel::Serializer
      def self.serializer_for(model, options)

Override the serializer_for method on MySerializer doesn't do anything, because the code always uses the definition of `serializer_for ` in SerializableResource not whatever model itself is referencing.
https://github.com/rails-api/active_model_serializers/blob/aad7779a3fd918bed5c17438c4626883d6fe6c5f/lib/active_model_serializers/serializable_resource.rb#L57

So I think the way it is currently structured, it is impossible to have custom serializers for polymorphic associations in a given parent serializer.

any thoughts?

**Pull request merged**